### PR TITLE
fix: Hooks gracefully return with an error message when optimizely prop is null

### PR DESCRIPTION
**Pull request by opti-jnguyen**
## Summary

Enables usage of hooks within the Optimizely Provider without passing in the `optimizely` prop.

Previously, attempting to use hooks within an Optimizely Provider with a missing `optimizely` prop led to an error being thrown resulting in a crash.

With this change, hooks now return pessimistic results when the `optimizely` prop is `null`.

Addresses this issue: 
- https://github.com/optimizely/react-sdk/issues/152

Also related:
- https://github.com/optimizely/react-sdk/issues/134

Note:
- Instead of returning an "Error" message like requested, we will defer to SDK users to handle the pessimistic values. 
- In certain edge cases, this may be a potentially breaking change for users that expect hooks to throw an error when the `optimizely` prop is null.

---

## Test Plan

Refactored tests that expect hooks to throw errors when Optimizely provider is not present.

### CODE REVIEW

File: src/Provider.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2018-2019, Optimizely
+ * Copyright 2022, Optimizely
```
Comment (opti-jnguyen) [inline src/Provider.tsx:2]: 2018-2019, 2022

File: src/hooks.ts [hunk 2]
```diff
@@ -365,11 +375,22 @@ export const useFeature: UseFeature = (featureKey, options = {}, overrides = {})
...
   const { optimizely, isServerSide, timeout } = useContext(OptimizelyContext);
+
+  const overrideAttrs = useCompareAttrsMemoize(overrides.overrideAttributes);
+
   if (!optimizely) {
-    throw new Error('optimizely prop must be supplied via a parent <OptimizelyProvider>');
+    hooksLogger.warn(`Unable to use decision ${flagKey}. optimizely prop must be supplied via a parent <OptimizelyProvider>`);
+    return [
+      createFailedDecision(flagKey, 'Optimizely SDK not configured properly yet.', {
+        id: overrides.overrideUserId || null,
```
Comment (opti-jnguyen) [inline src/hooks.ts:385]: Return null for user items bc user depends on Optimizely object

**Review by zashraf1985 — APPROVED** [review]
Looks Great! Can you change the PR description format to what we usually use like "Summary", "Test Plan" etc.
